src/rt/sections_elf_shared.d

```diff
@@ -711,12 +711,12 @@ void scanSegments(in ref dl_phdr_info info, DSO* pdso)
         case PT_LOAD:
             if (phdr.p_flags & PF_W) // writeable data segment
             {
-                auto beg = cast(void*)(info.dlpi_addr + phdr.p_vaddr);
+                auto beg = cast(void*)(info.dlpi_addr + (phdr.p_vaddr & ~(size_t.sizeof - 1)));
                 pdso._gcRanges.insertBack(beg[0 .. phdr.p_memsz]);
             }
             version (Shared) if (phdr.p_flags & PF_X) // code segment
             {
-                auto beg = cast(void*)(info.dlpi_addr + phdr.p_vaddr);
+                auto beg = cast(void*)(info.dlpi_addr + (phdr.p_vaddr & ~(size_t.sizeof - 1)));
                 pdso._codeSegments.insertBack(beg[0 .. phdr.p_memsz]);
```

Comment:
> Which I don't think it can ever be if (p_vaddr & p_align) == (p_offset & p_align) is to be satisfied.

@ibuclaw Could you add an explanation for why you can draw that conclusion as a code comment?